### PR TITLE
linux-wifi-hotspot: remove TKIP usage in create_ap

### PR DESCRIPTION
**Pull request by ARizzo35**
## Description of changes

Since [TKIP is now deprecated in hostapd](https://github.com/NixOS/nixpkgs/commit/4bec3f204362fa22a0740c8a572ffef3b322596d), the `create_ap` script in `linux-wifi-hotspot` needs to be updated to not use it.

## Things done

Changes were built and tested on aarch64-linux.

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/linux-wifi-hotspot/default.nix [hunk 1]
```diff
@@ -61,6 +61,13 @@ stdenv.mkDerivation rec {
       --replace "/usr" "$out"
   '';
 
+  # Since TKIP is deprecated in hostapd, we need to modify create_ap to not use it
+  # See: https://github.com/NixOS/nixpkgs/commit/4bec3f204362fa22a0740c8a572ffef3b322596d
+  preBuild = ''
```
Comment (Majiir) [inline pkgs/os-specific/linux/linux-wifi-hotspot/default.nix:66]: nit: Why `preBuild` and not `postPatch`?

**Review by Majiir — COMMENTED** [review]
What breaks when this patch isn't applied?

**Comment by oddlama** [comment]
> What breaks when this patch isn't applied?

Most likely hostapd will not start. Since this is generally an issue with linux-wifi-hotspot I've opened https://github.com/lakinduakash/linux-wifi-hotspot/pull/342 to track this upstream.

@ARizzo35 can you update this to use `patches = [(fetchpatch {...})]` to pull this in?

**Review by oddlama — APPROVED** [review]
LGTM

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 251191` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>linux-wifi-hotspot</li>
  </ul>
</details>

```
linux-wifi-hotspot> building '/nix/store/p05aslisk861xkks609wwi8xmp8z2i7d-linux-wifi-hotspot-4.6.0.drv'
linux-wifi-hotspot> unpacking sources
linux-wifi-hotspot> unpacking source archive /nix/store/7frg7g8qxim5qckdzj8z8d7dadgrh76i-source
linux-wifi-hotspot> source root is source
linux-wifi-hotspot> patching sources
linux-wifi-hotspot> applying patch /nix/store/niz8il406j5nvrxfl38ijxivrl7zlxxc-32fdb2bf2d3a34847270598682dec57cc3caf424.patch
linux-wifi-hotspot> patching file src/scripts/create_ap
linux-wifi-hotspot> Reversed (or previously applied) patch detected!  Assume -R? [n] 
linux-wifi-hotspot> Apply anyway? [n] 
linux-wifi-hotspot> Skipping patch.
linux-wifi-hotspot> 1 out of 1 hunk ignored -- saving rejects to file src/scripts/create_ap.rej
```

**Comment by onny** [comment]
this might be already fixed upstream?

**Comment by pbsds** [comment]
Seems the patch was tagged in 4.6.0 which git merged in #266989

**Pull request closed without merging**